### DISCUSSION TITLE
FR(sync): Evaluate passed revset to `stack(...)` w/ flag (or by default)

### DISCUSSION THREAD

**Discussion by claytonrcarter** [arxanas/git-branchless#1080] [Q&A] [answered]
I often use `git sync ...` as a quick short hand for `git move -b ... -d main` for individual branches. I have to explicitly pass `stack(...)` though, because `sync` seems to only want to move the exact commits in the passed revset. In fact, this behavior always leaves me wondering: is this intentional, and – if so – in what cases would you want to `sync` only specific commits?

**Proposal:** Would you be open to a change to the behavior of `sync` so that it would expand the passed revset to the equivalent of `stack(...)` either by default, or w/ a flag such as `-b/--base`? 
- if so, and if this becomes the default behavior, the current behavior would still be available via `git move`
- I'm not suggesting any changes to the behavior of `sync` when no args are passed. That would still sync all draft stacks.

**Comment by arxanas** [comment]
> is this intentional, and – if so – in what cases would you want to sync only specific commits?

It was intentional since I didn't know for certain how `git sync` would be used in practice. So far, I haven't had a use for the individual-commit-syncing use-case, and evidently, neither have you, so I think it's fine to make a breaking change for `git sync` to sync the entire `stack()` of the provided revset.

**Reply by claytonrcarter** [reply, marked as answer]
This is done via #1128
